catalog/gantt-chart/src/index.ts

```diff
@@ -222,21 +222,23 @@ Spotfire.initialize(async (mod) => {
                 color = node.rows()[0].color().hexCode;
             }
 
-            if (isCategorical) {
-                const colorValue = node.rows()[0].categorical("Color").value()[0].key;
-                const differentElements = node.rows().filter((r) => {
-                    return r.categorical("Color").value()[0].key !== colorValue;
-                }).length;
-                if (differentElements > 0) {
-                    color = config.defaultBarColor;
-                }
-            } else {
-                const colorValue = node.rows()[0].continuous("Color").value();
-                const differentElements = node.rows().filter((r) => {
-                    return r.continuous("Color").value() !== colorValue;
-                }).length;
-                if (differentElements > 0) {
-                    color = config.defaultBarColor;
+            if (hasColorExpression) {
+                if (isCategorical) {
+                    const colorValue = node.rows()[0].categorical("Color").value()[0].key;
+                    const differentElements = node.rows().filter((r) => {
+                        return r.categorical("Color").value()[0].key !== colorValue;
+                    }).length;
+                    if (differentElements > 0) {
+                        color = config.defaultBarColor;
+                    }
+                } else {
+                    const colorValue = node.rows()[0].continuous("Color").value();
```

Comment:
Do I understand things correctly that we can't simply compare the hexCodes across the rows, as some rows might be marked?

Reply:
That is also my interpretation. This pull request did not address any of the code inside this section though. It merely allows the color expression to be empty, and in such the proper color for a non leaf node is determined in rows 219-223.